src/extensions/extension_manager/src/extension_manager.cpp

```diff
@@ -740,36 +738,31 @@ ADUC_Result ExtensionManager::Download(
         goto done;
     }
 
-    if (!isFileExistsAndValidHash)
+    try
     {
-        try
-        {
-            result = downloadProc(entity, workflowId, workFolder, retryTimeout, downloadProgressCallback);
-        }
-        catch (...)
+        result = downloadProc(entity, workflowId, workFolder, retryTimeout, downloadProgressCallback);
+    }
+    catch (...)
+    {
+        result = { .ResultCode = ADUC_Result_Failure,
+                    .ExtendedResultCode = ADUC_ERC_CONTENT_DOWNLOADER_DOWNLOAD_EXCEPTION };
+        goto done;
+    }
+
+    if (IsAducResultCodeSuccess(result.ResultCode))
+    {
+        if (!ADUC_HashUtils_IsValidFileHash(
+                childManifestFile.str().c_str(),
+                ADUC_HashUtils_GetHashValue(entity->Hash, entity->HashCount, 0),
```

Comment:
Since GetHashValue can return NULL in case entity Hash/HashCount was corrupt for some reason (buffer overrun, etc), would be good to call it before and set result + goto done rather than passing NULL into IsValidFilehash.

Also, add `/* index */` after the zero value last arg.

Reply:
Fixed